src/Enzo/enzo_EnzoIntegrableUpdate.hpp

```diff
@@ -40,79 +39,80 @@ class EnzoIntegrableUpdate : public PUP::able
+  /// integrable quantities or are specified with `passive_list`. All elements
+  /// in these arrays are set to `value`.
   ///
-  /// @param block holds data to be processed
-  /// @param dUcons_group The grouping of fields to be modified. These fields
-  ///     are nominally used to accumulate the changes to all integrable and
-  ///     passively advected quantites.
-  /// @param value the value to assign to all elements of the fields
-  void clear_dUcons_group(Block *block, Grouping &dUcons_group,
-			  enzo_float value) const;
+  /// @param[in,out] dUcons_map The map holding the arrays that are to be
+  ///     modified. These arrays are nominally used to accumulate the changes
+  ///     to all integrable and passively advected quantites.
+  /// @param[in]     value The value array elements are assigned.
+  /// @param[in]     passive_list A list of keys for passive scalars.
+  void clear_dUcons_map(EnzoEFltArrayMap &dUcons_map, enzo_float value,
+                        const str_vec_t &passive_list) const noexcept;
 
   /// Computes the change in (the conserved form of) the integrable and
   /// passively advected quantites due to the flux divergence along dimension
   /// `dim` over the timestep `dt`. These changes are added to the accumulator
-  /// fields specified by `dUcons_group`.
+  /// arrays contained by `dUcons_map`.
   ///
-  /// @param block holds data to be processed
-  /// @param dim The dimension along which to compute the flux divergence
-  /// @param dt The current timestep.
-  /// @param flux_group Grouping of fields that hold the fluxes computed for
+  /// @param[in]     dim The dimension along which to compute the flux
+  ///     divergence.
+  /// @param[in]     dt The current timestep.
+  /// @param[in]     cell_width The cell width along dimension `dim`.
+  /// @param[in]     flux_map Map of arrays holding the fluxes computed for
   ///     the current timestep. The values of these fields should be stored
   ///     on the cell faces along the `dim` dimension.
-  /// @param dUcons_group The grouping of fields that the flux divergence is
-  ///     added to.
-  /// @param stale_depth The stale depth at the time of the function call.
-  ///     This should match the stale depth at the time the fluxes were
+  /// @param[in,out] dUcons_map Map of arrays where the flux divergence is added
+  ///     to. If constrained transport is being used, this will not include
+  ///     arrays for the magnetic fields.
+  /// @param[in]     stale_depth The stale depth at the time of the function
+  ///     call. This should match the stale depth at the time the fluxes were
   ///     computed.
-  void accumulate_flux_component(Block *block, int dim, double dt,
-				 Grouping &flux_group, Grouping &dUcons_group,
-				 int stale_depth) const;
+  /// @param[in]     passive_list A list of keys for passive scalars.
+  void accumulate_flux_component(int dim, double dt, enzo_float cell_width,
```

Comment:
@pgrete also made a similar suggestion. The use of `double` is very intentional here because `Method` subclass instances **always** compute the timestep (via the virtual `Method::timestep (Block * block) const` method) in double precision. This is entirely independent of the precision of `enzo_float`.

Within the context of `EnzoMethodMHDVlct`, I've tried to be self-consistent about always storing the (partial) timestep in double precision. I think changing that is somewhat beyond the scope of this PR (if we want to change it, I think we should make sure that all `Method` objects handle it self-consistently).